src/cmds/categories/create.js

```diff
@@ -0,0 +1,121 @@
+const chalk = require('chalk');
+const config = require('config');
+const changeCase = require('change-case');
+const { getProjectVersion } = require('../../lib/versionSelect');
+const fetch = require('../../lib/fetch');
+const { cleanHeaders, handleRes } = require('../../lib/fetch');
+const { debug } = require('../../lib/logger');
+
+module.exports = class CategoriesCreateCommand {
+  constructor() {
+    this.command = 'categories:create';
+    this.usage = 'categories:create <title> [options]';
+    this.description = 'Create a category with the specified title and guide in your ReadMe project';
+    this.category = 'categories';
+    this.position = 2;
+
+    this.hiddenargs = ['title'];
+    this.args = [
+      {
+        name: 'key',
+        type: String,
+        description: 'Project API key',
+      },
+      {
+        name: 'version',
+        type: String,
+        description: 'Project version',
+      },
+      {
+        name: 'title',
+        type: String,
+        defaultOption: true,
+      },
+      {
+        name: 'categoryType',
+        type: String,
+        description: 'Category type, must be guide or reference',
+      },
+    ];
+  }
+
+  async run(opts) {
+    const { categoryType, title, key, version } = opts;
+    debug(`command: ${this.command}`);
+    debug(`opts: ${JSON.stringify(opts)}`);
+
+    if (!key) {
+      return Promise.reject(new Error('No project API key provided. Please use `--key`.'));
+    }
+
+    if (!title) {
+      return Promise.reject(new Error(`No title provided. Usage \`${config.get('cli')} ${this.usage}\`.`));
+    }
+
+    if (categoryType !== 'guide' && categoryType !== 'reference') {
+      return Promise.reject(new Error('`categoryType` must be guide or reference.'));
+    }
+
+    const selectedVersion = await getProjectVersion(version, key, false);
+
+    debug(`selectedVersion: ${selectedVersion}`);
+
+    const slug = changeCase.paramCase(title);
+
+    function getNumberOfPages() {
+      return fetch(`${config.get('host')}/api/v1/categories?perPage=20&page=1`, {
+        method: 'get',
+        headers: cleanHeaders(key, {
+          'x-readme-version': selectedVersion,
+          Accept: 'application/json',
+        }),
+      }).then(res => {
+        return Math.ceil(res.headers.get('x-total-count') / 20);
+      });
+    }
+
+    const allCategories = [].concat(
+      ...(await Promise.all(
+        Array.from({ length: await getNumberOfPages() }, (_, i) => i + 1).map(async page => {
+          return fetch(`${config.get('host')}/api/v1/categories?perPage=20&page=${page}`, {
```

Comment:
If we do keep this slug check logic intact, I think we should look for the slug via [this endpoint](https://docs.readme.com/reference/getcategory) instead of fetching every category and iterating through all of them.

Reply:
@kanadgupta I thought about this, but ran into some issues. For Example:

1. Developer 1 uses the `categories:create` command to create a category with a title of `Test` and a type of `reference`. 
2. This would create a category with a slug of `test` and a type of `reference`
3. Developer 1 uses the `categories:create` command to  create a category with a title of `Test` and a type of `guide`
4. This would create a category with a slug of `test-1` and a type of `guide`
5. Developer 2 uses the `categories:create` command to  create a category with a title of `Test` and a type of `guide`
6. This would use [this endpoint](https://docs.readme.com/reference/getcategory) to search for an existing category with a slug of `test`, but because the category with a type of `guide`  in step 3 has a slug of `test-1` than a category will not be returned and a duplicate category of `Test` and a type of `guide` will be created with a slug of `test-2`.

Because the slug has an automatically incremented suffix, the user using the command will have to know that the suffix exists to search properly and to do that they will need to call [this endpoint](https://docs.readme.com/reference/getcategories) to determine that anyway.